actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -56,8 +60,11 @@ def collection_by_cache_keys(view, template)
 
         digest_path = view.digest_path_from_template(template)
 
+        @ordered_keys = []
```

Comment:
Instead of building this instance variable here and using in other methods we should return it and pass around.